src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -374,16 +374,15 @@ export default class DefaultDeviceController
   }
 
   async startVideoInput(device: VideoInputDevice): Promise<MediaStream | undefined> {
+    return await this.videoInputTaskQueue.add(() => this.startVideoInputTask(device));
```

Comment:
- Do I understand correctly that builders cannot handle an error anymore? i.e., The start input API does not throw an error.
- `await` only in the start API?

Reply:
NVM. Using `await` will let `add` throw an error so builders can handle it accordingly.

Reply:
I discussed this offline with Trung. The APIs behave like the following:

```js
try {
  await startVideoInput(deviceId) // Suppose that this call throws an error.

  // Not reached
  await startVideoInput(deviceId2)
} catch (error) {
  // Catch and handle an error.
}
```

```js
try {
  startVideoInput(deviceId) // Suppose that this call throws an error.

  // Reached
  startVideoInput(deviceId2)
} catch (error) {}
```

---

Only one concern we discussed is that if the browser's permission dialog appears, the queue will hold subsequent promise objects for a long time, confusing builders.